filament/backend/src/opengl/OpenGLProgram.cpp

```diff
@@ -92,7 +92,28 @@ float u16tofp32(highp uint v) {
 vec2 unpackHalf2x16(highp uint v) {
     return vec2(u16tofp32(v&0xFFFFu), u16tofp32(v>>16u));
 }
-
+uint fp32tou16(float val) {
+    uint f32 = floatBitsToUint(val);
+    uint f16 = 0u;
+    uint sign = (f32 >> 16) & 0x8000u;
+    int exponent = int((f32 >> 23) & 0xFFu) - 127;
+    uint mantissa = f32 & 0x007FFFFFu;
+    if (exponent > 15) {
+        f16 = sign | (0x1Fu << 10);
+    } else if (exponent > -15) {
+        exponent += 15;
+        mantissa >>= 13;
+        f16 = sign | uint(exponent << 10) | mantissa;
+    } else {
+        f16 = sign;
+    }
+    return f16;
+}
+highp uint packHalf2x16(vec2 v) {
```

Comment:
Can we use a built-in when it's available?

Reply:
Well, it should inject that code only on GL 4.1 when the extension is not available.

Reply:
Ah right, sorry the diff on GitHub wasn't showing enough context.